astropy/timeseries/periodograms/base.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from astropy.timeseries import TimeSeries, BinnedTimeSeries
+
+__all__ = ['BasePeriodogram']
+
+
+class BasePeriodogram:
+
+    @classmethod
+    def from_timeseries(cls, timeseries, *, column=None, error=None, **kwargs):
+        """
+        Initialize a periodogram from a time series object.
+
+        If a binned time series is passed, the time at the center of the bins is
+        used. Also note that this method automatically gets rid of NaN/undefined
+        values when initalizing the periodogram.
+
+        Parameters
+        ----------
+        column : str, optional
```

Comment:
I'm not a fan of the name `column` as it feels a bit ambiguous.  At a minimum I would prefer `column_name`, but what about something a bit more descriptive like `signal_column_name` or just `signal_name`?

Reply:
Fixed, and made it so it can be given as a positional argument too